homeassistant/core.py

```diff
@@ -1750,18 +1755,33 @@ def async_register(
         self,
         domain: str,
         service: str,
-        service_func: Callable[[ServiceCall], Coroutine[Any, Any, None] | None],
+        service_func: Callable[
+            [ServiceCall], Coroutine[Any, Any, ServiceResult] | None
+        ],
         schema: vol.Schema | None = None,
+        result_schema: vol.Schema | None = None,
```

Comment:
Should we validate the result? That seems like an overhead that might not be needed.

Reply:
Discussed in DM, we are going to remove result_schema. We do want to include it in services.yaml, but that should not be validated on execution.

Reply:
Removed.